Model/Observer/RecommendSettings.php

```diff
@@ -0,0 +1,216 @@
+            $this->configWriter->save($changedPath, 0);
+            throw new LocalizedException(__($e->getMessage()));
+        }
+    }
+
+    /**
+     * @param string $changedPath
+     * @return void
+     * @throws LocalizedException
+     */
+    protected function validateRelatedProducts(string $changedPath): void
+    {
+        try {
+            $recommendations = $this->recommendManagement->getRelatedProductsRecommendation($this->getProductId());
+            if (empty($recommendations['renderingContent'])) {
+                $this->configWriter->save($changedPath, 0);
+                throw new LocalizedException(__(
+                    "It appears that there is no trained model available for Related Products recommendation for the AppID: %1.",
+                    $this->configHelper->getApplicationID()
+                ));
+            }
+        } catch (\Exception $e) {
+            $this->configWriter->save($changedPath, 0);
+            throw new LocalizedException(__($e->getMessage()));
+        }
+    }
+
+    /**
+     * @param string $changedPath
+     * @return void
+     * @throws LocalizedException
+     */
+    protected function validateTrendingItems(string $changedPath): void
+    {
+        try {
+            $recommendations = $this->recommendManagement->getTrendingItemsRecommendation();
+            // When no recommendations suggested, most likely trained model is missing
+            if (empty($recommendations['renderingContent'])) {
+                $this->configWriter->save($changedPath, 0);
+                throw new LocalizedException(__(
+                    "It appears that there is no trained model available for Trending Items recommendation for the AppID: %1.",
+                    $this->configHelper->getApplicationID()
+                ));
+            }
+        } catch (\Exception $e) {
+            $this->configWriter->save($changedPath, 0);
+            throw new LocalizedException(__($e->getMessage()));
+        }
+    }
+
+    /**
+     * @param string $changedPath
+     * @return void
+     * @throws LocalizedException
+     */
+    protected function validateLookingSimilar(string $changedPath): void
+    {
+        try {
+            $recommendations = $this->recommendManagement->getLookingSimilarRecommendation($this->getProductId());
+            if (empty($recommendations['renderingContent'])) {
+                $this->configWriter->save($changedPath, 0);
+                throw new LocalizedException(__(
+                    "It appears that there is no trained model available for Looking Similar Recommendation for the AppID: %1.",
+                    $this->configHelper->getApplicationID()
+                ));
+            }
+        } catch (\Exception $e) {
+            $this->configWriter->save($changedPath, 0);
+            throw new LocalizedException(__($e->getMessage()));
+        }
+    }
+
+    /**
+     * @return string
+     */
+    private function getProductId(): string
```

Comment:
Just curious - Why are we casting to a string? Is this because the API endpoint expects `objectID` as a string?

Reply:
Yes, objectID accepts string

Reply:
Cool -  thanks for clarifying. Good use of casting here. 👍